lib/ansible/vault.rb

```diff
@@ -12,6 +14,22 @@ module Ansible
   class Vault
     # The standard header for Ansible's current vault format
     FILE_HEADER = "$ANSIBLE_VAULT;1.1;AES256".freeze
+    # YAML_KEY_VALUE_REGEX is used to guess whether a file is a YAML or not
+    YAML_KEY_VALUE_REGEX = /^(\w+): !vault \|$/.freeze
+
+    # Indicate if the text at the supplied path appeard to be encrypted by
+    # Ansible Vault
+    #
+    # @param text [String] The encrypted text
+    # @param password [String] The password for the text
+    def self.decrypt(text:, password:)
+      matches = YAML_KEY_VALUE_REGEX.match(text)
+      if matches && !matches.empty?
```

Comment:
@sunakan @tpickett66 I'm open to ideas how to better detect if a file is YAML or not. :thinking:

Reply:
Done as you suggested.

I've "deprecated" `decrypt`. It still uses `decrypt_text` but it can be removed in next major version as per https://semver.org/#how-should-i-handle-deprecating-functionality